joomla/joomla-cms
# [4.1] Child templates 2/2 [clean]

Pull Request for Issue # .


### Summary of Changes
- Added functionality for the creation of a child template (for templates that support it). The name convention for the child template is [parent template]_[child template name] (eg for cassiopeia's child named temp the actual name will be `cassiopeia_temp`
- Tweaked the language inclusions so children templates of the core get a correct translation (templates that have the language folder were already covered)
- Tweaked the functionality for the preview thumb and preview images. Also added a fallback image if there's not an image in the template...
- Tweaked the file editor so that folders could be created/deleted/renamed both on the template folder or the media folder
- Tweaked the file editor so that files could be edited/created/deleted/renamed/uploaded/extracted both on the template folder or the media folder

### Testing Instructions
- Git fetch this PR, run npm
- Try every functionality in the com_templates for both core templates

- Install https://github.com/joomla/joomla-cms/files/7235154/cassy_1.0.0.zip (cassiopeia with a different name)
- Try every functionality in the com_templates for the installed template

- Also check (with tinymce as the editor) when editing an article that the editor.css (or .min.css if not in debug mode) is loaded from the `/css` folder of the template, if such file exists. (the path can be `/templates/templateName/css/editor.[min.]CSS` for existing templates or  `media/templates/site/templateName/css/editor.[min.]CSS` for templates supporting children).

### Actual result BEFORE applying this Pull Request



### Expected result AFTER applying this Pull Request



### Documentation Changes Required

Yes, this is the UI for the codename `child template` or whatever the project ends up naming the inheritable/extendable/... templates
